WordPress/src/main/java/org/wordpress/android/ui/sitecreation/NewSiteCreationMainVM.kt

```diff
@@ -32,32 +45,41 @@ data class SiteCreationState(
 typealias NavigationTarget = WizardNavigationTarget<SiteCreationStep, SiteCreationState>
 
 class NewSiteCreationMainVM @Inject constructor() : ViewModel() {
-    private val wizardManager: WizardManager<SiteCreationStep> = WizardManager(
-            // TODO we'll receive this from a server/Firebase config
-            listOf(
-                    SiteCreationStep.fromString("site_creation_segments"),
-                    SiteCreationStep.fromString("site_creation_verticals"),
-                    SiteCreationStep.fromString("site_creation_site_info"),
-                    SiteCreationStep.fromString("site_creation_domains"),
-                    SiteCreationStep.fromString("site_creation_site_preview")
-            )
-    )
     private var isStarted = false
-    private var siteCreationState = SiteCreationState()
-
-    val navigationTargetObservable: SingleEventObservable<NavigationTarget> = SingleEventObservable(
-            Transformations.map(wizardManager.navigatorLiveData) {
-                WizardNavigationTarget(it, siteCreationState)
-            }
-    )
+    private lateinit var wizardManager: WizardManager<SiteCreationStep>
+    private lateinit var siteCreationState: SiteCreationState
+
+    val navigationTargetObservable: SingleEventObservable<NavigationTarget> by lazy {
+        SingleEventObservable(
+                Transformations.map(wizardManager.navigatorLiveData) {
+                    WizardNavigationTarget(it, siteCreationState)
+                }
+        )
+    }
 
     private val _wizardFinishedObservable = SingleLiveEvent<CreateSiteState>()
     val wizardFinishedObservable: LiveData<CreateSiteState> = _wizardFinishedObservable
 
-    fun start() {
+    fun start(savedInstanceState: Bundle?) {
         if (isStarted) return
+        if (savedInstanceState == null) {
+            siteCreationState = SiteCreationState()
+            wizardManager = WizardManager(SITE_CREATION_STEPS)
+        } else {
+            siteCreationState = savedInstanceState.getParcelable(KEY_SITE_CREATION_STATE)
+            val currentStepIndex = savedInstanceState.getInt(KEY_CURRENT_STEP)
+            wizardManager = WizardManager(SITE_CREATION_STEPS, currentStepIndex)
+        }
         isStarted = true
-        wizardManager.showNextStep()
+        if (savedInstanceState == null) {
+            // Show the next step only if it's a fresh activity so we can handle the navigation
+            wizardManager.showNextStep()
+        }
```

Comment:
Can the `wizardManager.showNextStep()` statement be moved under the `if (savedInstanceState == null) {` branch [here](https://github.com/wordpress-mobile/WordPress-Android/pull/8892/files#diff-0fd7e9e72406631ad0d31dd1d3e03e45R65) or is it intentionally called after `isStarted = true`?

Reply:
It's intentionally called after `isStarted = true`. It wouldn't make a difference where it's called right now, but there is no guarantee that it won't in the future and it's good to mark things as `ready` before we handle the navigation.

Reply:
I don't think setting the `isStarted` as `true` early is the correct approach. We should set it after we know the setup is done. It's currently not used as a check but it _can_ be and if we set it early, there is a chance it could be read before the setup is done from another thread.

I can move the `wizardManager.showNextStep()` to previous `if`, if you feel strongly about it, but I personally believe the current version is the best approach.